File: Day285/moveAlZerosToEndOfArray.go
package main

import (
	"fmt"
)

func pushZerosToEnd(arr []int, n int) {
	zero := 0
	for i:=0; i<n; i++ {
		if arr[i] != 0 {
			temp := arr[zero]
			arr[zero] = arr[i]
			arr[i] = temp
			zero++
		}

	}
}

func main() {
	n1 := 5
	n2 := 4
	arr1 := []int{3,5, 0, 0, 4}
	arr2 := []int{0, 0, 0, 4}
	
	pushZerosToEnd(arr1, n1)
	pushZerosToEnd(arr2, n2)
	fmt.Println(arr1)
	fmt.Println(arr2)
}

File: Day285/moveAlZerosToEndOfArray.java
package Day285;

import java.util.Arrays;

public class moveAlZerosToEndOfArray {
    static void pushZerosToEnd(int[] arr, int n) {
        // code here
        int zero = 0;
        for (int i=0; i<arr.length; i++) {
            if (arr[i] != 0) {
                int temp = arr[zero];
                arr[zero++] = arr[i];
                arr[i] = temp;
                
            }
        }
       
    }
    public static void main(String[] args) {
        int n1 = 5;
        int n2 = 4;
        int arr1[] = {3,5, 0, 0, 4};
        int arr2[] = {0, 0, 0, 4};
            
        pushZerosToEnd(arr1, n1);
        pushZerosToEnd(arr2, n2);
        System.out.println(Arrays.toString(arr1));
        System.out.println(Arrays.toString(arr2));
    }
}